### ISSUE TITLE
Regression: a module named `std` has already been imported in this module [E0252]

### ISSUE THREAD

**Issue by briansmith** [rust-lang/rust#30089]
See https://travis-ci.org/briansmith/ring/jobs/91080822

Relevant build output:
```
src/file_test.rs:17:5: 17:8 error: a module named `std` has already been imported in this module [E0252]
src/file_test.rs:17 use std;
                        ^~~
src/file_test.rs:17:5: 17:8 help: run `rustc --explain E0252` to see a detailed explanation
src/file_test.rs:16:1: 16:14 note: previous import of `std` here
src/file_test.rs:16 use super::*;
```

This code built fine on Nightly last week. It still build correctly on Beta and Release; see https://travis-ci.org/briansmith/ring/builds/91080819.

The code in question is:

```
use rustc_serialize::hex::FromHex;
use super::*;
use std;
use std::io::BufRead;
```

You can try it by cloning https://github.com/briansmith/ring, revision ee3ff646962e4a4933eaabb50c76f196521fd592.

**Comment by briansmith** [comment]
Replacing the `use super::*;` with `use super::digest;` worked around the issue.

**Comment by alexcrichton** [comment]
Minimized:

```rust
mod foo {        
    use std;     
    use super::*;
}                
fn main() {}                
```

Happened between 2ba44607fe09c4fb98a52ca2149b0c9869a87b4c (2015-11-23 nightly) and  a9d15a6e92f2f68dad632b20ad26d3698dec75f8 (2015-11-24 nightly), the cluprit of which is likely #29530.

cc @nikomatsakis, @jseyfried, perhaps something leaked in by accident? Either that or perhaps that's the wrong PR?

**Comment by nikomatsakis** [comment]
Yes, seems very likely that this is caused by that PR. @jseyfried, please let me know if you have time to look into this. I fear I do not have cycles to spare to find a more tailored fix, so otherwise we may want to just pull the PR and rebase it. That said, the problem is presumably some interaction with the injection of the standard prelude, and I guess the (implicit) `extern crate` that is added there?

**Comment by jseyfried** [comment]
#29530 caused it, sorry about that. I explained and fixed the regression in PR #30133.

**Comment by jseyfried** [comment]
cc @alexcrichton @nikomatsakis
